### DISCUSSION TITLE
Testing thermal runaway

### DISCUSSION THREAD

**Discussion by saskford** [Jyers/Marlin#1221] [Q&A] [answered]
I used the USB method to test if the thermal runaway was working. After several minutes the USB adapter started to get pretty warm trying to heat up the nozzle. The temperature was rising extremely slowly but the alarm never went off. I then downloaded an old Creality firmware and reran the test and the alarm did go off fairly quickly. I have v1.3.4 installed. Is there a better way to test thermal runaway or is it possible that it is not turned on in this version?

**Comment by Jyers** [comment, marked as answer]
Thermal Runaway is 100% enabled on this firmware. If you really want to test it you really want to test you could take out your thermistor and try heating the nozzle.

If you want the specific specifications, in v1.3.4 thermal runaway will trip if the nozzle fails to heat up by 3 degrees within 20 seconds, this has been tuned to 30 seconds in the latest source to reduce accidental trips.

**Reply by jimbc** [reply]
Hi,
I can also cannot activate the thermal runaway error. I tried every method I know of. (1) usb plugged in, AC plugged out and set nozzle temp. to 200 degrees, (2) heat with AC plugged in to 200 degrees, cut off power and remain USB power, (3) took the thermistor out as proposed and set the nozzle to 200 degrees for 2 mins. I stopped it manually as it started to produce smoke and (4) unplugged wiring for heater, set nozzle to 200 degrees and let it run for 5 mins. No error occurred for any of those tests. I'm used [E3V2-BLTouch-5x5-HS-v4.2.2-v2.0.1.bin](https://github.com/Jyers/Marlin/releases/download/v2.0.1/E3V2-BLTouch-5x5-HS-v4.2.2-v2.0.1.bin) for my Ender 3 V2. 
If this is a mistake in the settings, please tell me what I need to configure, otherwise I consider the thermal runaway pretection not working for now.

**Reply by saskford** [reply]
Another way to check if it is working is to plug a cell phone charger into the usb socket on the printer with the printer turned off. When your screen comes on set the nozzle temp to 200. Since the charger doesn't have enough power to run the heater a thermal issue will be detected and the printer will give you an error message. In my case this happens in less than a minute. Here is a link to another full featured firmware I found that you could use until this problem is fixed in Jyer's version. It solved the problem on my E3 V2. 
https://github.com/mriscoc/Ender3V2S1/wiki

**Comment by jimbc** [comment]
Hi @saskford,
Your method is what I tried to explain by method (1). I retried it and waited 10 mins this time without success. On stock firmware I get an alarm after around 4.5 mins, which I also consider too slow as PLA starts to smoke after less than 3 mins.

Hi @tititopher68-dev,
I just installed the latest release and cannot confirm it is working with all above mentioned method. What do I need to do?

**Reply by saskford** [reply]
Apparently Marlin added a feature that was supposed to make the feature more sensitive. Unfortunately it has caused it to not work altogether on at least some E3 V2 printers. Some people are blaming it on poor grounding of the motherboard done at the factory. The ground connection tested perfect on my printer but I double grounded it anyway but thermal runaway shutdown still does not work. The new feature can be turned off when the firmware is compiled but in this case he has not chosen to do so. I think a lot of people are in danger of a fire as a result and don't even know it. 4 to 5 minutes is pretty slow alright. On other firmware mine shuts down in less than 1 minute when tested doing the USB method. I absolutely love the firmware that he has created but right now it is just too big of a risk for me.

**Reply by jimbc** [reply]
@saskford 
Do you know which line I need to comment off to deactivate the new feature or activate to get it into running again?

**Reply by saskford** [reply]
This line needs to be disabled.

THERMAL_PROTECTION_VARIANCE_MONITOR

I wish I had the ability to do this so I could keep using this firmware.
